plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeLegacyCreateTableWithExistingLocation.java

```diff
@@ -0,0 +1,116 @@
+
+public abstract class BaseDeltaLakeLegacyCreateTableWithExistingLocation
+        extends AbstractTestQueryFramework
+{
+    private static final Logger LOG = Logger.get(BaseDeltaLakeLegacyCreateTableWithExistingLocation.class);
+
+    protected static final String CATALOG_NAME = "test_delta";
+    protected static final String SCHEMA = "legacy_create_table_with_existing_location_" + randomTableSuffix();
+
+    protected File metastoreDir;
+    protected HiveMetastore metastore;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = createTestSession();
+
+        metastoreDir = Files.createTempDirectory("test_delta_lake").toFile();
+        FileHiveMetastoreConfig config = new FileHiveMetastoreConfig()
+                .setCatalogDirectory(metastoreDir.toURI().toString())
+                .setMetastoreUser("test");
+        metastore = new FileHiveMetastore(
+                new NodeVersion("testversion"),
+                HDFS_ENVIRONMENT,
+                false,
+                config);
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
+        queryRunner.installPlugin(new TestingDeltaLakePlugin());
+
+        queryRunner.createCatalog(CATALOG_NAME, CONNECTOR_NAME, getConnectionProperties());
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+
+        return queryRunner;
+    }
+
+    protected abstract Session createTestSession();
+
+    protected abstract Map<String, String> getConnectionProperties();
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        try {
+            if (metastore != null) {
+                // Data is on the local disk and will be deleted by the deleteOnExit hook
+                metastore.dropDatabase(SCHEMA, false);
+                deleteRecursively(metastoreDir.toPath(), ALLOW_INSECURE);
+            }
+        }
+        catch (Exception e) {
+            LOG.error(e, "Failed to clean up Glue database: %s", SCHEMA);
+        }
+    }
+
+    @Test
+    public void testLegacyCreateTable()
+    {
+        String tableName = "test_legacy_create_table_" + randomTableSuffix();
+
+        assertQuerySucceeds(format("CREATE TABLE %s AS SELECT 1 as a, 'INDIA' as b, true as c", tableName));
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true')");
+
+        String tableLocation = (String) computeScalar("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '') FROM " + tableName);
+        String tableNameNew = tableName + "_new";
+
+        assertQuerySucceeds(format("CREATE TABLE %s.%s.%s (dummy int) with (location = '%s')", CATALOG_NAME, SCHEMA, tableName + "_new", tableLocation));
+
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true')");
```

Comment:
Remove redundant `ROW` (including other places) and just use `1` & `true` (type names are redundant).

Reply:
Addressed!